xla/hlo/experimental/auto_sharding/auto_sharding_strategy.h

```diff
@@ -42,7 +43,7 @@ constexpr double kInfinityCost = 1e20;
 
 // Type alias
 template <typename Key, typename Value>
-using StableHashMap = ::absl::flat_hash_map<Key, Value>;
+using StableHashMap = ::absl::btree_map<Key, Value>;
```

Comment:
Please add a test that checks that `StableHashMap` now indeed has deterministic order. E..g a test where you insert 10 pairs in a random order and expect to read them in a known order.

Reply:
TBH I think this is fine, we never had flat_hash_map here before (it's different internally), and not everything is deterministic yet, so this fix clearly makes it incrementally better.

Reply:
I'm not opposed to the solution, just want to have a test that clearly states that whatever `StableHashMap` is defined to be, it should be deterministic.